packages/react-components/react-infobutton/src/components/InfoLabel/useInfoLabel.ts

```diff
@@ -55,6 +56,7 @@ export const useInfoLabel_unstable = (props: InfoLabelProps, ref: React.Ref<HTML
 
   if (infoButton) {
     infoButton['aria-labelledby'] ??= `${label.id} ${infoButton.id}`;
+    root['aria-owns'] = (infoButton.info as InfoButtonProps).id;
```

Comment:
You'll need to resolve the shorthand for the info slot, and generate this ID:

```ts
const infoButtonInfo = resolveShorthand(infoButton.info, {
  defaultProps: {
   id: generatedInfoId, // from a useId call outside this if statement
  }
});
infoButton.info = infoButtonInfo;
root['aria-owns'] = infoButtonInfo.id;
```

Reply:
Also, you don't want to overwrite any existing aria-owns prop, Wrap that all with `if (root['aria-owns'] === undefined) {`